uat/testing-features/src/main/resources/greengrass/features/log-manager-1.feature

```diff
@@ -228,9 +228,57 @@ Feature: Greengrass V2 LogManager
            }
         }
         """
-        And the local Greengrass deployment is SUCCEEDED on the device after 60 seconds
+        And the local Greengrass deployment is SUCCEEDED on the device after 180 seconds
         When device network connectivity is online
         Then I verify that it created a log group of type GreengrassSystemComponent for component System, with streams within 60 seconds in CloudWatch
         Then I verify that it created a log group of type UserComponent for component UserComponentY, with streams within 60 seconds in CloudWatch
         And I verify 20 logs for UserComponentY of type UserComponent have been uploaded to Cloudwatch within 80 seconds
 
+    Scenario: LogManager-1-T5: When more than 10_000 component logs are to be uploaded, then log manager uploads all
+    the logs in multiple attempts where each attempt has a maximum of 10_000 log events.
+
+        Given I create a log directory for component called UserComponentWLogDirectory
+        And I create a Greengrass deployment with components
+            | aws.greengrass.Cli        | LATEST                                    |
+            | aws.greengrass.LogManager | classpath:/greengrass/recipes/recipe.yaml |
+        When I update my Greengrass deployment configuration, setting the component aws.greengrass.LogManager configuration to:
+        """
+        {
+            "MERGE": {
+                "logsUploaderConfiguration": {
+                    "componentLogsConfiguration": [
+                        {
+                            "logFileRegex": "UserComponentW(.*).log",
+                            "logFileDirectoryPath": "${UserComponentWLogDirectory}",
+                            "componentName": "UserComponentW"
+                        }
+                    ],
+                    "systemLogsConfiguration": {
+                        "uploadToCloudWatch": "true",
+                        "minimumLogLevel": "INFO",
+                        "diskSpaceLimit": "25",
+                        "diskSpaceLimitUnit": "MB",
+                        "deleteLogFileAfterCloudUpload": "true"
+                    }
+                },
+                "periodicUploadIntervalSec": "300"
```

Comment:
why is this set to 300? Shouldn't you set this very very low so the test can execute more quickly?

Adding this UAT seems to have increased the OTF time from 7 minutes to over 20!

Reply:
As soon as the custom component writing logs is deployed, LM starts uploading active logs (which will be less than 10K) because it is deployed first. This is to force LM to wait and upload 10K+ logs at once by giving enough time for the custom component to be deployed (~3 min) and write 10K+ logs. 240 seconds may also work.

Reply:
why don't you just deploy the custom component first then?